### PR TITLE
Add UTF-8 sanitizer to Strings plugin

### LINKED ISSUE #9113: SNMP Trap input does not properly handle non-string data
### System info:

1.18.1 running in docker

### Steps to reproduce:

When using the snmp_trap input plugin, some devices send traps with non utf-8 encoded strings, for instance:

```
snmpTrapEnterprise.0="jnxProductNameEX4500" sysUpTimeInstance="3297275066" jnxCmRescueChgTime.0="3297275065" jnxCmRescueChgDate.0="�	&+" jnxCmRescueChgSource.0="2" jnxCmRescueChgUser.0="user" jnxCmRescueChgState.0="2" 
```

### Expected behavior:

See https://github.com/influxdata/telegraf/issues/9113#issuecomment-829585208

### Actual behavior:

See https://github.com/influxdata/telegraf/issues/9113#issuecomment-829585208

### PR DESCRIPTION
**Pull request by loganmc10**
### Required for all PRs:

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.

Adds a new option for the Strings processer to sanitize strings so that they conform to utf-8

### CODE REVIEW

**Review by Jagularr — APPROVED** [review]
this makes sense to me!

**Review by ssoroka — APPROVED** [review]

File: plugins/processors/strings/README.md [hunk 1]
```diff
@@ -78,6 +79,11 @@ If you'd like to apply multiple processings to the same `tag_key` or `field_key`
   ## Decode a base64 encoded utf-8 string
   # [[processors.strings.base64decode]]
   #   field = "message"
+
+  ## Sanitize a string to ensure it is a valid utf-8 string. Non-valid characters are replaced by 'new'
+  # [[processors.strings.valid_utf8]]
+  #   field = "message"
+  #   new = ""
```
Comment (jessingrass) [inline plugins/processors/strings/README.md:86]: Could you change `new` to `replacement`? New does not describe what is happening in this code.

File: plugins/processors/strings/README.md [hunk 2]
```diff
@@ -78,6 +79,11 @@ If you'd like to apply multiple processings to the same `tag_key` or `field_key`
   ## Decode a base64 encoded utf-8 string
   # [[processors.strings.base64decode]]
   #   field = "message"
+
+  ## Sanitize a string to ensure it is a valid utf-8 string. Non-valid characters are replaced by 'new'
```
Comment (jessingrass) [inline plugins/processors/strings/README.md:83]: Could you clarify your description to align more closely with the go documentation around ToValidUTF8 function docs? Here's a link to the relevant documentation and description: https://golang.org/pkg/strings/#ToValidUTF8

**Review by jessingrass — CHANGES_REQUESTED** [review]

**Comment by loganmc10** [comment]
Ok I've incorporated that feedback

**Comment by sspaink** [comment]
!retry-checks

**Comment by sspaink** [comment]
@loganmc10 not sure why the CI didn't trigger for this pull request. Would you be able to push an empty commit to see if it would trigger it? You can use `git commit -m 'empty commit' --allow-empty`

**Comment by loganmc10** [comment]
@sspaink ok done, it does look a little more normal now? I never saw this workflow approval gate before

**Pull request merged**